eth/stagedsync/stage_mining_create_block.go

```diff
@@ -34,10 +34,18 @@ type MiningBlock struct {
 	RemoteTxs types.TransactionsStream
 }
 
+// In case we are proposing during proof-of-stake and we are supplied with header fields already
+type PresetHeaderFields struct {
+	Timestamp             uint64
+	Random                common.Hash
+	SuggestedFeeRecipient common.Address
+}
+
 type MiningState struct {
 	MiningConfig    *params.MiningConfig
 	PendingResultCh chan *types.Block
 	MiningResultCh  chan *types.Block
+	ProposeBlockCh  chan *types.Block
```

Comment:
Why not reuse `MiningResultCh`? If the idea is to have a separate channel for PoS blocks, let's call it `PoSMiningResultCh` or something.

Reply:
because it might be used for something else, also i would not name it Mining because we are not mining, we are proposing

Reply:
which is a similar, yet fundamentally different process

Reply:
OK, fair enough, but ProposeBlockCh still lives inside MiningState :) To my mind it's fine to call PoS payload building mining, but it's not that important.